src/main/typescript/components/form/DepositForm.tsx

```diff
@@ -81,17 +82,41 @@ const DepositForm = (props: DepositFormProps) => {
     const formState = useSelector(state => state.depositForm)
     const fileState = useSelector(state => state.files.loading)
     const fileUploadInProgress = useSelector(isFileUploading)
-    const formValues = useSelector(state => state.form.depositForm && state.form.depositForm.values)
+    const formValues = useSelector(state => state.form.depositForm?.values)
     const dispatch = useDispatch()
 
     const doSave = () => {
         const shouldSetToDraft = props.depositState.label === DepositStateLabel.REJECTED
         formValues && dispatch(saveDraft(props.depositId, formValues, shouldSetToDraft))
     }
+    const doAutoSave = () => {
+        // TODO what about REJECTED deposits?
+        //  Should they be set to DRAFT as well on autosave?
+        //  Or should autosave not happen as long as they're in REJECTED state?
+        if (props.depositState.label !== DepositStateLabel.REJECTED && formValues)
+            dispatch(autoSaveDraft(props.depositId, formValues))
+    }
     const doSubmit: (data: DepositFormMetadata) => void = data => {
         const shouldSetToDraft = props.depositState.label === DepositStateLabel.REJECTED
         dispatch(submitDeposit(props.depositId, data, history, shouldSetToDraft))
     }
+
+    const cachedFormValues = useRef(formValues)
+    useEffect(() => {
+        cachedFormValues.current = formValues
+    }, [(formValues && Object.keys(formValues).length !== 0)])
+    const autoSaveAfterMS = inDevelopmentMode
+        ? 10000 // 10 seconds in development mode
+        : 300000 // 5 minutes in production mode
+    const debouncedFormValues = useDebounce(formValues, autoSaveAfterMS)
+    useEffect(() => {
+        if (!isEqual(cachedFormValues.current, formValues)) {
+            console.log("auto save triggered")
+            cachedFormValues.current = formValues
+            doAutoSave()
```

Comment:
what happens when you click save after an edit? is auto-save gonna be triggered in that situation?